src/consensus/aft/raft.h

```diff
@@ -2640,14 +2671,22 @@ namespace aft
       }
     }
 
-    void become_leader()
+    void become_leader(bool force_become_leader = false)
     {
       if (is_retired())
       {
         return;
       }
 
-      election_index = last_committable_index();
+      if (consensus_type == ConsensusType::BFT && !force_become_leader)
```

Comment:
Is the special case for force_become_leader necessary? I would have thought the initial rollback seqno could be returning the right initial value for election index. If that's the case, then last_committable_index() can be doing the switching on BFT/CFT, and this call can stay unchanged and avoid the flag.

Reply:
I added a comment to the code to explain this